### PR TITLE
#680 impl(call to layer intersect service)

### PR DESCRIPTION
**Pull request by qifeng-bai**
PR for #680 
What it does:
1, Submit layer intersection request to Spatial service
2, Check the task status
3, Read sample.csv in zip file from the given downloaded url.

Concerns:
1, Is ZipStream closed properly
2, What is the proper TIMEOUT for the task
3, What is the 'status', if the task fails

### CODE REVIEW

File: src/main/java/au/org/ala/biocache/service/AlaLayersService.java [hunk 1]
```diff
@@ -239,7 +245,83 @@ public String getLayersServiceUrl() {
...
+                        ResponseEntity<String> statusResponse = restTemplate.getForEntity(statusUrl, String.class);
+                        if (statusResponse.getStatusCode() == HttpStatus.OK) {
+                            JSONObject processResult = JSONObject.fromObject(statusResponse.getBody());
+
+                            String status = processResult.getString("status");
+                            if (status.equalsIgnoreCase("finished")) {
+                                String downloadUrl = processResult.getString("downloadUrl");
+                                logger.debug("Reading intersect result: " + downloadUrl);
+                                URL url = new URL(downloadUrl);
+                                HttpURLConnection connection = (HttpURLConnection) url.openConnection();
```
Comment (adam-collins) [inline src/main/java/au/org/ala/biocache/service/AlaLayersService.java:282]: Connection not closed anywhere.

File: src/main/java/au/org/ala/biocache/stream/ProcessDownload.java [hunk 2]
```diff
@@ -250,21 +250,24 @@ private void intersectAnalysisLayers() {
...
-                            if (sampling != null && sampling.length == headers.analysisIds.length + 2) {
-                                // suitable space is already available in each batch row String[]
-                                System.arraycopy(sampling, 2, batch.get(j), headers.labels.length, sampling.length - 2);
+                    if (reader != null) {
+                        CSVReader csv = new CSVReader(reader);
+                        intersection = csv.readAll();
+                        csv.close();
+
+                        for (int j = 0; j < batch.size(); j++) {
+                            if (j > batch.size()) {
```
Comment (adam-collins) [inline src/main/java/au/org/ala/biocache/stream/ProcessDownload.java:259]: Remove `if` statement that is always false.

**Review by adam-collins — CHANGES_REQUESTED** [review]
2 changes required and apply formatting rules.

File: src/main/java/au/org/ala/biocache/stream/ProcessDownload.java (hunk 2 above)
Comment (qifeng-bai) [reply src/main/java/au/org/ala/biocache/stream/ProcessDownload.java:259]: If spatial service does not return 'completed' status,  Null reader will be returned.

File: src/main/java/au/org/ala/biocache/service/AlaLayersService.java (hunk 1 above)
Comment (qifeng-bai) [reply src/main/java/au/org/ala/biocache/service/AlaLayersService.java:282]: Yes, that one is tricky. The solution I can figure out is to store zip to local temp folder and close the connection, and then go to next step

**Comment by adam-collins** [comment]
obsolete

**Pull request closed without merging**